bigbluebutton-html5/imports/ui/components/app/component.jsx

```diff
@@ -37,7 +37,13 @@ const propTypes = {
   sidebar: PropTypes.element,
   media: PropTypes.element,
   actionsbar: PropTypes.element,
+  closedCaption: PropTypes.element,
+  userList: PropTypes.element,
+  chat: PropTypes.element,
   locale: PropTypes.string,
+  intl: PropTypes.shape({
```

Comment:
Should use the `intlShape`

Reply:
Importing a prop type object from the package to validate itself? That's a joke, right?

This prop validation is supposed to show you what changed and where in the code. But it has zero value if we use intlShape, since they'll change the object together with the API update.

Reply:
https://github.com/yahoo/react-intl/wiki/API#intlshape

> to make sure its of the correct object-shape, React Intl v2 has an intlShape module export. Here's how you access and use the API:

https://github.com/yahoo/react-intl/issues/162

Reply:
> This prop validation is supposed to show you what changed and where in the code. But it has zero value if we use intlShape, since they'll change the object together with the API update.

PropTypes are for typechecking. They are not mean to be used to show what/where something changed in the code. They are mean to be used to check if you are passing the correct type to a prop, if the component in this case is expecting a `intl` prop with the shape of the `intl` object provided by the `react-intl` package, makes TOTAL sense to be using their `intlShape`.

https://reactjs.org/docs/typechecking-with-proptypes.html

> As your app grows, you can catch a lot of bugs with typechecking. For some applications, you can use JavaScript extensions like Flow or TypeScript to typecheck your whole application.

Reply:
> PropTypes are for typechecking.

What do we want to check in the type of the `intl` prop in this case? We want to check if the object passed have the shape provided by the `react-intl` package.

I understand your concern, but It will only happen if the package (`react-intl`) have a major update with breaking changes (i.e. changing the `formatMessage` name or signature) and using typechecking to prevent that is actually a bad idea imo. Major updates are not meant to be backward compatible.

The typechecking in this case imo is just to assure the component that he has the `intl` prop as it expects with all the functionality provided by `react-intl`.

Reply:
> We want to check if the object passed have the shape provided by the react-intl package.
That is exactly the mistake that you are making.

Our component expects the props to be in a certain format, and that's what we are checking here - whether the props meet **our** component's expectations. Otherwise it will just break.

What you are suggesting is to ask `react-intl` if the object we imported is the object from `react-intl`.
It sounds really strange, doesn't it?

Reply:
> Our component expects the props to be in a certain format

Surprise, surprise, it happens to expect exactly `intlShape` :sweat_smile:

Reply:
The component happens to expect what it uses in the code, not what is specified in the `intlShape`, since it can change by a package.
Not sure what your point is here. You just found matching properties?

Reply:
Think in a strongly-typed programming language (Java, Typescript, ...). Our "class" `AppComponent` is expecting a parameter `intl` to be a instance of a class/implementation of `ReactIntl`.
Even if the `AppComponent` class ins't using all powers from the `ReactIntl` instance it still expects the full instance of that object. 

Otherwise, why aren't we passing the `formatMessage` only? Makes no sense to pass the full object, if the component is only using a single function. (:

Reply:
It doesn't matter if they expect/use a full instance, or just one method.

The component makes sure that it receives props in the format that it knows how to use.
But the whole point of this will be lost if we ask `react-intl` to validate itself. Guess what? It will never fail PropTypes checking in this case 😄

Reply:
And it shouldn't fail. It should fail if we don't pass a `react-intl` instance because that is what the component is expecting.

Reply:
Or you think passing something like `intl={{ formatMessage: () => { console.log('i don't understand typecheckign' } }}` is ok for the component?